src/Tools/Source/RunTests/Program.cs

```diff
@@ -43,33 +43,54 @@ internal static int Main(string[] args)
                 return ExitFailure;
             }
 
+            if (options.CollectDumps)
+            {
+                if (!DumpUtil.IsAdministrator())
+                {
+                    ConsoleUtil.WriteLine(ConsoleColor.Yellow, "Dump collection specified but user is not administrator so cannot modify registry");
+                }
+                else
+                {
+                    DumpUtil.EnableRegistryDumpCollection(options.LogFilesDirectory);
+                }
+            }
+
             // Setup cancellation for ctrl-c key presses
-            var cts = new CancellationTokenSource();
+            using var cts = new CancellationTokenSource();
             Console.CancelKeyPress += delegate
             {
                 cts.Cancel();
+                DisableRegistryDumpCollection();
             };
 
-            var result = Run(options, cts.Token).GetAwaiter().GetResult();
+            int result;
+            if (options.Timeout is { } timeout)
+            {
+                result = await RunAsync(options, timeout, cts.Token);
+            }
+            else
+            {
+                result = await RunAsync(options, cts.Token);
+            }
+
             CheckTotalDumpFilesSize();
+            DisableRegistryDumpCollection();
```

Comment:
Is there any concern where something here can throw an exception and then this doesn't get reset? Should it be in a finally block?

Reply:
I went back and forth on this. The part my brain got stuck on is that the process can crash or be killed hence there is no way to guarantee this is done. Then looking at our pre-existing code there was no attempt at all to undo the registry change. 

I can use the `finally` block though. My brain just got stuck on some "how correct can this really be" logic.

Reply:
There is no need to reset this. Machine state following a ci build is undefined.

Reply:
True when this is invoked from CI. I'm trying to treat this as a general option though, one which just happens to always be enabled when run from CI. Hence the need to reset.